float32-herbie/src/float32.c

```diff
@@ -23,8 +23,42 @@ float mul_f32(float a, float b)
     return a * b;
 }
 
-// Avoid segfaulting
+// Avoid error
 float div_f32(float a, float b)
 {
     return (b == 0.0f) ? NAN : (a / b);
+}
+
+int f32_to_ordinal(float a)
+{
+    int *pa;
+
+    if (a < 0.0f)
+    {
+        a = 0.0f - a;
+        pa = (int*)(&a);
+        return -(*pa);
+    }
+    else
+    {
+        pa = (int*)(&a);
+        return *pa;
```

Comment:
I'm concerned about 1) NaNs, 2) Negative zero, 3) undefined behavior. We need to be very careful with these. We actually had an implementation of this that we tested very carefully, years ago for the first Herbie paper. We might be able to dig that out of the git history... Here: https://github.com/uwplse/herbie/blob/1d8a5a266b020440095bcd8cb501c635b072ad95/infra/bruteforce.c